Spec.md

```diff
@@ -246,6 +256,26 @@ spec:
       # AWS Region to be used for the provider
       region: eu-central-1
 
+# AzureKV configures this store to sync secrets using Azure Key-Vault provider
+    azurekv:
+      # Auth defines the information necessary to authenticate against Azure by
+      # getting the Service-Princpal's clientID and clientSecret from an already created Kubernetes Secret
+      auth:
+        # The Azure Tenant to send requests to.
+        tenantId: 4be10619-c5d4-4032-bd6a-a697cb365a4a
```

Comment:
Generally LGTM. I think we should be consistent with naming the fields. In particular the ones that contain an `id`; should they be:
(1) `tenantId` / `accessKeyIdSecretRef` (aka `strict camelCase`)
(2) `tenantID` / `accessKeyIDSecretRef` (aka `golang style camelCase` )

From the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions):
> JSON field names must be camelCase. Other than capitalization of the initial letter, the two should almost always match.

Well, this leaves room for interpretation. :smile: 

I went through a bit of code and found:
* `cert-manager` uses golang-style
* k8s core volume APIs (`awsElasticBlockStore.volumeID`, `photonPersistentDisk.podID`, `portworxVolume.volumeID` etc.) use golang-style
* a lot of other repositories are inconsistent (either with api conventions or with their id naming)

My gut feeling says `tenantId` but after doing the research i'd opt in for `tenantID` . What do you think?

Any thoughts @knelasevero @mcavoyk ?

Reply:
Agreed with strict camelCase (Azure's JSON mostly follow this too): The PR is being updated to reflect the above.

Reply:
Azure's native JSON is strict camelCase, so following it is closer to native Azure payloads. I know it does not matter that much given that mapping in GoLang is explicit (unlike Node.js), but adhering to the same format may clear potential mis-understanding (IMHO).

Reply:
> closer to native Azure payloads

agreed, that's a valid point :+1: . CR validation will give the user immediate feedback if there's a typo. Let's stay with `tenantId`

Reply:
Fine by me :)